### ISSUE TITLE
[2.2.2] break the last pynitrokey update

### ISSUE THREAD

**Issue by tuxmaster5000** [Nitrokey/nitrokey-app2#227]
When updating pynitrokey in an venv environment which contain both, nitrokeyapp and pynitrokey it will fails with:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
nitrokeyapp 2.2.2 requires pynitrokey==0.4.45, but you have pynitrokey 0.4.46 which is incompatible.
```

**Comment by daringer** [comment]
yes, this makes sense, guess `nkapp2` is simply too strictly pinnned to its matching `pynitrokey` version - let's see if we can improve for the next release

**Comment by robin-nitrokey** [comment]
Relaxing the constraint would most likely lead to the opposite problem, Nitrokey App 2 not being compatible with newer pynitrokey releases.  The proper fix would be to extract the library parts from pynitrokey and use semver for it.  Until we have done that, I would prefer to keep the pinned pynitrokey version in nitrokey-app2.  This can always be worked around by installing pynitrokey into a separate venv or via pipx (which is what we recommend anyway).

**Comment by robin-nitrokey** [comment]
Just to be clear, we could of course relax it to `">=0.4.45, <0.4.47"` in the next release because we know that 0.4.46 is compatible with 0.4.45.  But I would only do that if we do a release anyway and not make extra releases for that, and only for already released pynitrokey versions that are known to be compatible.

**Comment by daringer** [comment]
closing, as this shouldn't happen anymore using nitrokey-sdk (?) please re-open if you see similar issues
